templates/_helpers.tpl

```diff
@@ -124,17 +139,13 @@ imagePullPolicy: {{ .root.Values.image.pullPolicy }}
 env:
   {{- $configMapNameOverride := .root.Values.configMapNameOverride }}
   {{- range $name, $value := .container.env }}
-    {{- if eq ( default "value" $value.type ) "value" }}
-    - name: {{ $name | quote }}
-      value: {{ $value.value | quote }}
-    {{- else }}
-    {{- if ne $value.type "none" }}
-    - name: {{ $name | quote }}
-      valueFrom:
-        {{ $value.type }}KeyRef:
-          name: {{ default $value.name ( get $configMapNameOverride $value.name ) | quote }}
-          key: {{ $value.key | quote }}
+    {{- if not ( default false $value.last ) }}
+      {{- include "common.oneEnv" ( dict "name" $name "value" $value "configMapNameOverride" $configMapNameOverride ) | indent 4 -}}
     {{- end }}
+  {{- end }}
+  {{- range $name, $value := .container.env }}
+    {{- if default false $value.last }}
+      {{- include "common.oneEnv" ( dict "name" $name "value" $value "configMapNameOverride" $configMapNameOverride ) | indent 4 -}}
```

Comment:
also, what happens if somebody puts `last` more than once? The order will be random for all the keys that have `last: true` isn't it?

Reply:
We also have an attribute last to be able to use the $(env) syntax...
This syntax gets only the environment variable declared before...

It's ordered alphabetically, but I suppose that we don't need to use recursive template values ...

Reply:
I don't understand, so basically you want to include env variable into other env variables at runtime? Or is this $(env_var) rendered at templating time?

Reply:
In middle time :-P, it's done by Kubernetes

Reply:
https://kubernetes.io/docs/tasks/inject-data-application/define-interdependent-environment-variables/#define-an-environment-dependent-variable-for-a-container

Reply:
interesting,
maybe you should put more logic? Like a numeric field `order: <int>`  that gets written down in the correct order so that you can chain that, instead of on boolean field `last` . I mean, one could put something like:
```yaml
VAL_A:
  type: value
  value: blabla$(VAL_B)
 order: 2
VAL_B:
  type: value
  value: something_interesting
 order: 1
```

Reply:
Initially, It's what I want to do, but I didn't find a way to order a list with helm :-/

Reply:
=> last true => order > 0